internal/app/machined/pkg/runtime/v1alpha1/bootloader/sdboot/sdboot.go

```diff
@@ -116,20 +116,22 @@ func ProbeWithCallback(disk string, options options.ProbeOptions, callback func(
 				return err
 			}
 
+			// here we handle a case when we boot of just kernel+initrd/uki and we don't have a booted entry
+			// then we know we're booted of UKI
+			if bootedEntry == "" && len(ukiFiles) == 1 {
+				sdbootConf = &Config{}
```

Comment:
I'm a bit confused here, should we just set `bootedEntry` to `filepath.Base(ukiFiles[0])`, and the loop below will set the config's Default?